### PR TITLE
Ensure header gets written first time when opening file

### LINKED ISSUE #2046: Add header to log.csv file
### Problem Description

_As an engineer debugging logs, it would be helpful to know what each column meant._

### Expected behavior

The log.csv file should have headers (a label for each column). Currently it doesn't.

### Additional context

Headers seem to be specified but for some reason aren't working
https://github.com/sdv-dev/SDV/blob/29cf34110867b87464b252177b8a59a6240d1c20/sdv/logging/logger.py#L16-L18

### PR DESCRIPTION
**Pull request by lajohn4747**
resolves #2046 
CU-86b0rhymd

Confirmed that for a new log file will have the header only once:

```
LEVEL,EVENT,TIMESTAMP,SYNTHESIZER CLASS NAME,SYNTHESIZER ID,TOTAL NUMBER OF TABLES,TOTAL NUMBER OF ROWS,TOTAL NUMBER OF COLUMNS
INFO,Instance,2024-06-11 11:47:55.245786,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_a36c8ccec65d40fca204ca1afe2baf3d,,,
INFO,Fit,2024-06-11 11:47:55.245925,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_a36c8ccec65d40fca204ca1afe2baf3d,1,500,9
INFO,Fit processed data,2024-06-11 11:47:55.344018,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_a36c8ccec65d40fca204ca1afe2baf3d,1,500,6
INFO,Sample,2024-06-11 11:47:55.466236,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_a36c8ccec65d40fca204ca1afe2baf3d,1,10,9
INFO,Instance,2024-06-11 11:48:04.461437,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_a1cc97ff1dc74789a2e24e7754ffdadb,,,
INFO,Fit,2024-06-11 11:48:04.461563,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_a1cc97ff1dc74789a2e24e7754ffdadb,1,500,9
INFO,Fit processed data,2024-06-11 11:48:04.558117,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_a1cc97ff1dc74789a2e24e7754ffdadb,1,500,6
INFO,Sample,2024-06-11 11:48:04.682670,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_a1cc97ff1dc74789a2e24e7754ffdadb,1,10,9
INFO,Instance,2024-06-11 11:49:21.612837,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_2ce9505e272348fe8ed371b9817813db,,,
INFO,Fit,2024-06-11 11:49:21.613059,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_2ce9505e272348fe8ed371b9817813db,1,500,9
INFO,Fit processed data,2024-06-11 11:49:21.722263,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_2ce9505e272348fe8ed371b9817813db,1,500,6
INFO,Sample,2024-06-11 11:49:21.847287,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_2ce9505e272348fe8ed371b9817813db,1,10,9
INFO,Instance,2024-06-11 11:53:11.433892,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_f831515cfa1e4aa19e2064be948e6749,,,
INFO,Fit,2024-06-11 11:53:11.434110,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_f831515cfa1e4aa19e2064be948e6749,1,500,9
INFO,Fit processed data,2024-06-11 11:53:11.530300,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_f831515cfa1e4aa19e2064be948e6749,1,500,6
INFO,Sample,2024-06-11 11:53:11.652728,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_f831515cfa1e4aa19e2064be948e6749,1,10,9
INFO,Instance,2024-06-11 11:53:44.828702,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_4799f7cd7cd94614b6568bce604cf16d,,,
INFO,Fit,2024-06-11 11:53:44.828923,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_4799f7cd7cd94614b6568bce604cf16d,1,500,9
INFO,Fit processed data,2024-06-11 11:53:44.931025,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_4799f7cd7cd94614b6568bce604cf16d,1,500,6
INFO,Sample,2024-06-11 11:53:45.055451,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_4799f7cd7cd94614b6568bce604cf16d,1,10,9
INFO,Instance,2024-06-11 11:54:09.605423,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_06650191040a43b4a9b7276bae8d812d,,,
INFO,Fit,2024-06-11 11:54:09.605662,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_06650191040a43b4a9b7276bae8d812d,1,500,9
INFO,Fit processed data,2024-06-11 11:54:09.697771,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_06650191040a43b4a9b7276bae8d812d,1,500,6
INFO,Sample,2024-06-11 11:54:09.818474,GaussianCopulaSynthesizer,GaussianCopulaSynthesizer_1.13.2.dev0_06650191040a43b4a9b7276bae8d812d,1,10,9
```

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b0rhymd SDV - Add header to log.csv file #2046](https://app.clickup.com/t/86b0rhymd)

File: sdv/logging/logger.py [hunk 1]
```diff
@@ -10,14 +11,18 @@
 class CSVFormatter(logging.Formatter):
     """Logging formatter to convert to CSV."""
 
-    def __init__(self):
+    def __init__(self, filename=None):
```
Comment (amontanez24) [inline sdv/logging/logger.py:14]:
```suggestion
    def __init__(self, filepath=None):
```
I think this has to be the full path to the file right?

Comment (lajohn4747) [reply sdv/logging/logger.py:14]: Yeah it is the full path, just named `filename` in the handler

Comment (amontanez24) [reply sdv/logging/logger.py:14]: Ah ok

**Review by pvk-developer — APPROVED** [review]

**Pull request merged**